### PR TITLE
chore: add GitHub Actions workflow to test builds

### LINKED ISSUE #34: Add workflow to run checks on pull requests
Create Actions workflow to run formatting, linting, and build checks on pull requests to `main`.

### PR DESCRIPTION
**Pull request by KevinWu098**
## Summary
1. Use default Github Node.js action to validate build

Closes #34

### CODE REVIEW

File: .github/workflows/node.js.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [18.x]
+        # See supported Node.js release schedule at https://nodejs.org/en/about/releases/
```
Comment (taesungh) [inline .github/workflows/node.js.yml:20]: Suggestion: if we aren't testing against multiple versions, I feel using a matrix is a bit excessive: we could directly specify the version. Additionally, could we version 20 (currently active LTS) to match our [deployment workflow](https://github.com/designatuci/DUCI-website/blob/main/.github/workflows/deploy-main.yml)?

File: .github/workflows/node.js.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+    steps:
+    - uses: actions/checkout@v4
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v3
```
Comment (taesungh) [inline .github/workflows/node.js.yml:25]: Chore: could we use the latest version, actions/setup-node@v4

File: .github/workflows/node.js.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
+# This workflow will do a clean installation of node dependencies, cache/restore them, build the source code and run tests across different versions of node
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-nodejs
+
+name: Node.js CI
```
Comment (taesungh) [inline .github/workflows/node.js.yml:4]: Suggestion: While at the moment this workflow only builds the project, we could imagine wanting to run other linting, formatting, and testing scripts (see #34). Could we rename this workflow (and the filename) to something like "Run Checks"?

**Review by taesungh — COMMENTED** [review]
Thanks for looking into this, I meant to add a long time ago as part of #34 but never had the chance. Some small suggestions on naming and versions.

**Comment by KevinWu098** [comment]
Thanks for the feedback @taesungh ! I've linked #34 to this PR and added in linting as an additional check

File: .github/workflows/node.js.yml (hunk 3 above)
Comment (taesungh) [reply .github/workflows/node.js.yml]: Could we also rename the file to `run-checks.yml`

Comment (KevinWu098) [reply .github/workflows/node.js.yml]: done!

File: .github/workflows/run-checks.yml [hunk 4]
```diff
@@ -0,0 +1,32 @@
+# This workflow will do a clean installation of node dependencies, cache/restore them, build the source code and run tests across different versions of node
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-nodejs
+
```
Comment (taesungh) [inline .github/workflows/run-checks.yml:3]: Chore: comment is no longer accurate (particularly about different versions of node)

**Review by taesungh — APPROVED** [review]
Thanks for starting to add this, will hopefully help catch potential errors before merging. In the future, we can also look into more intricate actions which leave review comments.

**Pull request merged**